core/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/request/handler/aasserialization/GenerateSerializationByIdsRequestHandler.java

```diff
@@ -52,24 +58,33 @@ public GenerateSerializationByIdsRequestHandler(RequestExecutionContext context)
 
 
     @Override
-    public GenerateSerializationByIdsResponse process(GenerateSerializationByIdsRequest request) throws ResourceNotFoundException {
+    public GenerateSerializationByIdsResponse process(GenerateSerializationByIdsRequest request) throws ResourceNotFoundException, SerializationException {
+        DefaultEnvironment environment = new DefaultEnvironment.Builder()
+                .assetAdministrationShells(
+                        request.getAasIds().stream()
+                                .map(LambdaExceptionHelper.rethrowFunction(x -> context.getPersistence().getAssetAdministrationShell(x, OUTPUT_MODIFIER)))
+                                .collect(Collectors.toList()))
+                .submodels(request.getSubmodelIds().stream()
+                        .map(LambdaExceptionHelper.rethrowFunction(x -> context.getPersistence().getSubmodel(x, OUTPUT_MODIFIER)))
+                        .collect(Collectors.toList()))
+                .conceptDescriptions(request.getIncludeConceptDescriptions()
+                        ? context.getPersistence().findConceptDescriptions(
+                                ConceptDescriptionSearchCriteria.NONE,
+                                OUTPUT_MODIFIER,
+                                PagingInfo.ALL)
+                                .getContent()
+                        : List.of())
+                .build();
+        List<InMemoryFile> files = new ArrayList<>();
+        Map<String, FileContent> fileMap = context.getFileStorage().getAllFiles();
```

Comment:
If possible we should only include the files that are actually referenced by the selected elements. This probably can be done using visitors.